keyboard_handler/src/keyboard_handler_unix_impl.cpp

```diff
@@ -135,21 +155,27 @@ KeyboardHandlerUnixImpl::KeyboardHandlerUnixImpl(
   }
 
   struct termios new_term_settings;
-  if (tcgetattr_fn(stdin_fd_, &old_term_settings) == -1) {
+  if (tcgetattr_fn(stdin_fd_, &old_term_settings_) == -1) {
     throw std::runtime_error("Error in tcgetattr(). errno = " + std::to_string(errno));
   }
 
-  signal(SIGINT, quit);  // Setup signal handler to return terminal in original (buffered) mode
-  // in case of abnormal program termination.
+  if (install_signal_handler) {
+    old_sigint_handler_ = std::signal(SIGINT, on_signal);  // Setup signal handler to return
```

Comment:
There a few things that this class do that can be a problem if one creates multiple instances of it (even before this PR).
I think that in the future is worth considering having a "singlenton" instance, though that's a bit unrelated to this PR.

Reply:
There are prons and cons for applying singleton pattern for the `KeyboardHandler` class.
And I have even had it implemented as singleton on some early stage of `KeyboardHandler`development, but latter on decided to come back to the regular class.
Form one hand implementing `KeyboardHandler` as singleton could potentially solve some problems related to the concurrency and data integrity in case of existence of the multiple instances in one process.
From another hand singleton pattern assume that there will be only one call for constructor and one call for the destructor during all liveliness of the process. 
And latter is a problem. Because `KeyboardHandler` heavily rely on the callbacks. In case of the regular instance created on the stack or on the heap with smart pointer, all registered callbacks will be automatically cleaned up upon destructor call when `KeyboardHandler` instance goes out of scope. 
In case of the singleton, destructor will not be called and "dead" callbacks will persist in the callback list of the `KeyboardHandler` when some unexpected  exception happened in the application but process hasn't been finished but rather continue running from some place with recreating whole hierarchy of the classes. 
Callbacks which are pointing to the deleted objects are very dangerous since will cause code execution from unexpected memory pointer which doesn't belong any more too those callbacks.
And this is much worse rather than potential race conditions.
If think more thorough the `KeyboardHandler`is an entity which doesn't imply to exist for more than one instance per process and if such situation is possible by design it could be considered as severe design mistake in the application and could be pretty much easily overcome via wrapping out by mutex  on upper level of the application.